### PR TITLE
MISRA: fix rule 8.9 violations

### PR DESCRIPTION
**Pull request by alfred2g**
MISRA: fix rule 8.9 violations

Description
-----------
variables should be defined inside functions if used only in those functions

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/portable/BufferManagement/BufferAllocation_2.c [hunk 1]
```diff
@@ -84,7 +84,6 @@ static size_t uxMinimumFreeNetworkBuffers;
  * in this array.  The array is not accessed directly except during initialisation,
  * when the xFreeBuffersList is filled (as all the buffers are free when the system
  * is booted). */
```
Comment (shubnil) [inline source/portable/BufferManagement/BufferAllocation_2.c:86]: Please move the comment as well to line 98.

File: source/FreeRTOS_IP.c [hunk 2]
```diff
@@ -186,6 +186,9 @@ NetworkAddressingParameters_t xNetworkAddressing = { 0, 0, 0, 0, 0 };
 
 /** @brief Default values for the above struct in case DHCP
  * does not lead to a confirmed request. */
+
+/* used for unit testing */
```
Comment (AniruddhaKanhere) [inline source/FreeRTOS_IP.c:190]: One might ask - "How is this used for unit-testing?"
Can we maybe make the comment a bit more descriptive? I think we should take a page from FreeRTOS-Kernel about the comments.

**Review by AniruddhaKanhere — COMMENTED** [review]
Small NIT pick.

**Review by AniruddhaKanhere — APPROVED** [review]

File: source/FreeRTOS_TCP_IP.c [hunk 3]
```diff
@@ -64,6 +64,10 @@
 /* Just make sure the contents doesn't get compiled if TCP is not enabled. */
 #if ipconfigUSE_TCP == 1
 
+
+/* declared global for unit testing purposes */
+/* coverity[misra_c_2012_rule_8_9_violation] */
+
```
Comment (paulbartell) [inline source/FreeRTOS_TCP_IP.c:70]: This is static, but becomes a global during unit testing right?

**Review by paulbartell — APPROVED** [review]

**Pull request merged**